gtsfm/common/gtsfm_data.py

```diff
@@ -346,3 +346,43 @@ def filter_landmarks(self, reproj_err_thresh: float = 5) -> "GtsfmData":
                 filtered_data.add_track(track)
 
         return filtered_data
+
+    def align_via_Sim3_to_poses(self, wTi_list_ref: List[Optional[Pose3]]) -> "GtsfmData":
+        """Align estimated, sparse multiview result (self) to a set of reference poses.
+
+        Args:
+            wTi_list_ref: list of reference/target camera poses, ordered by camera index.
+
+        Returns:
+            aligned_data: sparse multiview result that is aligned to the poses above.
+        """
+        # these are the estimated poses (source, to be aligned)
+        wTi_list = self.get_camera_poses()
+
+        # align the poses which are valid (i.e. are not None)
+        # some camera indices may have been lost after pruning to largest connected component, leading to None values
+        wTi_list_aligned, gtSest = geometry_comparisons.align_poses_sim3_ignore_missing(wTi_list_ref, wTi_list)
+
+        aligned_data = GtsfmData(number_images=self.number_images())
+        # update the camera pose to the aligned poses, but use the previous calibration
+        for i, wTi in enumerate(wTi_list_aligned):
+            if wTi is None:
+                continue
+            calibration = self.get_camera(i).calibration()
+            aligned_data.add_camera(i, PinholeCameraCal3Bundler(wTi, calibration))
+
+        # align estimated tracks to the ground truth
+        for j in range(self.number_tracks()):
+            # align each 3d point
+            track_est = self.get_track(index=j)
+            # place into the GT reference frame
+            pt_gt = gtSest.transformFrom(track_est.point3())
+            track_aligned = SfmTrack(pt_gt)
+
+            # copy over the 2d measurements directly into the new track
+            for k in range(track_est.number_measurements()):
+                i, uv = track_est.measurement(k)
+                track_aligned.add_measurement(i, uv)
+            aligned_data.add_track(track_aligned)
```

Comment:
If we are following OOO style, the track should have a function to perform a similarity transform.

Reply:
I see, are you thinking we should add that to the `gtsam` SfmTrack struct, as a new method @ayushbaid ?

Reply:
Or maybe a separate utility function. Will be easier to unit test. 

Let's not go the GTSAM route as it will consume lot of time.

Reply:
I think the cleanest way to do it would be to add a function on Similarity(3) that operates on an SfmTrack, returning a new SfmTrack.